backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/ConvertDiskCommand.java

```diff
@@ -167,11 +167,21 @@ protected void executeCommand() {
             return;
         }
 
-        long requiredSize =  actionReturnValue.getActionReturnValue();
+        long requiredSize = actionReturnValue.getActionReturnValue();
+
+        CreateVolumeContainerCommandParameters parameters = createVolumeCreationParameters(getDiskImage(), requiredSize);
+        StorageDomain storageDomain = storageDomainDao.get(getLocationInfo().getStorageDomainId());
+
+        // Initial size will not be used for this configuration, meaning the required size will not
+        // be used at all. However, when converting to COW/Preallocated the required size may larger
+        if (parameters.getVolumeType() == VolumeType.Preallocated &&
+                storageDomain.getStorageType().isBlockDomain()) {
+            parameters.setSize(Math.max(getDiskImage().getSize(), requiredSize));
+            log.info("Updated disk size to: '{}'", parameters.getSize());
```

Comment:
if we log this as `Updated` then better to check if getDiskImage().getSize() != requiredSize first